server/main.steps.ts

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+import http from 'http';
+import https from 'https';
+import { resolve, sep } from 'path';
+import { writeFile, mkdir } from 'fs';
+import { ChildProcess, exec } from 'child_process';
+import {
+  Given,
+  When,
+  Then,
+  Before,
+  After,
+  Fusion,
+  And,
+} from 'jest-cucumber-fusion';
+import {
+  genericWorldType,
+  worldGenerator,
+} from '../test_common/jest_cucumber_support/commonTestTypes';
+
+import { devEnvToUseTls } from '../utils/tooling/secureDevUtil';
```

Comment:
Is this dynamic per tests? Or something you expect to change infrequently in config?

Reply:
Dynamic per run. All this is doing is seeing if the certs generated by openssl are present. If they are, they are used. It does make the HTTPS test here moot if not, but there is a clear warning printed if so. I expect CI will always have certs available